service/storage_service.cc

```diff
@@ -6023,6 +6023,14 @@ future<locator::load_stats> storage_service::load_stats_for_tablet_based_tables(
         });
     }
 
+    // Retrieval of tablet stats must be serialized with mutation to token metadata,
+    // as the former requires tablet id stability.
+    // If tablet split is finalized while retrieving stats, the saved erm, used by all
+    // shards, can have a lower tablet count than the one in a particular shard, causing
+    // an abort as tablet map requires that any id feeded into it is lower than its
+    // current tablet count.
+    auto tmlock = co_await get_token_metadata_lock();
```

Comment:
Why not pass the current tablet_map to the filter function instead?

Reply:
tablet_id is meaningful within a given tablet_map version, so we should use that version.

Reply:
the reason for having a global erm is that we want consistent view across shards (busy shards might be processing a tablet map different than the ones processed earlier). with intra node migration (which can complete fast), we could have double accounting. 

why if we still take token metadata lock (as done in this patch) and feed local tablet map into filter function (as you suggest)? then we can get rid of this global erm approach.

Reply:
Across-node stats collection is still async, so it only helps with intra-node migration. But ok.

I think this is orthogonal to tablet id stability, and we should still pass tablet_map& to the handler for simplicity.

Reply:
sure, will proceed with the changes.